packages/drawer/src/DrawerToolbarLayout/DrawerToolbarLayoutContainer.tsx

```diff
@@ -1,88 +1,68 @@
 import { EmbeddedDrawerLayout } from '../EmbeddedDrawerLayout';
 import { OverlayDrawerLayout } from '../OverlayDrawerLayout';
+import { DEFAULT_LGID_ROOT, getLgIds } from '../utils';
 
-import { DrawerToolbarLayoutContainerProps } from './DrawerToolbarLayout.types';
-
-// Dummy content for now
-const LongContent = () => {
-  return (
-    <div
-      className={css`
-        display: flex;
-        flex-direction: column;
-        gap: ${spacing[100]}px;
-      `}
-    >
-      <Body>
-        Ipsam aspernatur sequi quo esse recusandae aperiam distinctio quod
-        dignissimos. Nihil totam accusamus. Odio occaecati commodi sapiente
-        dolores.
-      </Body>
-      <Body>
-        Molestiae accusantium porro eum quas praesentium consequuntur deleniti.
-        Fuga mollitia incidunt atque. Minima nisi fugit sapiente.
-      </Body>
-      <Body>
-        Ratione explicabo saepe occaecati. Et esse eveniet accusamus veritatis
-        esse quod. Vero aliquid quasi saepe vel harum molestiae rerum.
-      </Body>
-      <Body>
-        Minima distinctio eligendi sit culpa tempore adipisci. Consequuntur
-        consequatur minus quaerat sapiente consectetur esse blanditiis
-        provident. Nulla quas esse quasi a error sint pariatur possimus quia.
-      </Body>
-    </div>
-  );
-};
+import {
+  DrawerToolbarLayoutContainerProps,
+  LayoutData,
+} from './DrawerToolbarLayout.types';
 
 export const DrawerToolbarLayoutContainer = ({
   children,
-  displayMode = DisplayMode.Embedded,
   data,
   onClose,
+  displayMode = DisplayMode.Embedded,
+  'data-lgid': dataLgId = DEFAULT_LGID_ROOT,
   darkMode: darkModeProp,
 }: DrawerToolbarLayoutContainerProps) => {
   const { darkMode } = useDarkMode(darkModeProp);
-  const [open, setOpen] = useState(false);
+  const { openDrawer, closeDrawer, getActiveDrawerContent, shouldCloseDrawer } =
+    useDrawerToolbarContext();
+  const isDrawerOpen = shouldCloseDrawer;
+  const { id, title, content } = getActiveDrawerContent() || {};
+  const lgIds = getLgIds(dataLgId);
+
+  // If there is no data, we don't want to render anything
+  if (!data || data.length === 0) return null;
 
   const handleOnClose = (event: React.MouseEvent<HTMLButtonElement>) => {
     onClose?.(event);
-    setOpen(false);
-
-    // TODO: add call to context to update the drawer content
+    closeDrawer();
   };
 
+  // If the display mode is overlay, we want to use the OverlayDrawerLayout, else we want to use the EmbeddedDrawerLayout
   const LayoutComponent =
     displayMode === DisplayMode.Overlay
       ? OverlayDrawerLayout
       : EmbeddedDrawerLayout;
 
+  // We want to pass the isDrawerOpen prop to the LayoutComponent only if the display mode is embedded
   const layoutProps = {
     hasToolbar: true,
-    isDrawerOpen: displayMode === DisplayMode.Embedded ? open : false,
+    isDrawerOpen: displayMode === DisplayMode.Embedded ? isDrawerOpen : false,
```

Comment:
Nit: I believe this is logically equivalent to just

```suggestion
    isDrawerOpen: displayMode === DisplayMode.Embedded && isDrawerOpen,
```